### PR TITLE
feat: support custom SSL certificates in SQL data sources

### PR DESCRIPTION
**Pull request by palpatim**
## Problem

Enables end-to-end support for custom SSL certificates in SQL data sources by tying together all of:
- https://github.com/aws-amplify/amplify-category-api/pull/2615
- https://github.com/aws-amplify/amplify-category-api/pull/2631
- https://github.com/aws-amplify/amplify-api-next/pull/237

Docs update: https://github.com/aws-amplify/docs/pull/7802

## New public API

Data schema configuration now includes the ability to define an `sslCert` secret.

```typescript
const schema = configure({
  database: {
    engine: 'mysql',
    connectionUri: secret('SQL_CONNECTION_STRING'),
    // v----- Added
    sslCert: secret('CUSTOM_SSL_CERT'),
  },
});
```

The `sslCert` secret will be resolved at deploy time and passed to the CDK construct `sslCertConfig` setting added in the [`@aws-amplify/data-construct` 1.9.0 release](https://github.com/aws-amplify/amplify-category-api/releases/tag/%40aws-amplify%2Fdata-construct%401.9.0).

## Changeset

This change introduces a new feature API. The only code changes are in `@aws-amplify/backend-data`. There are explicit dependency updates in both `@aws-amplify/backend-data` and `@aws-amplify/schema-generator`:
- `@aws-amplify/backend-data`
- `@aws-amplify/schema-generator`

## Validation

- Unit tests
- New unit tests included
- Manual testing

## Checklist

- [x] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [x] If this PR requires a docs update, I have linked to that [docs PR](https://github.com/aws-amplify/docs/pull/7802).
- [x] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "",
   "type": "module",
   "scripts": {
-    "build": "tsc --build packages/* && tsc --build scripts",
+    "build": "tsc --build packages/* scripts",
```
Comment (palpatim) [inline package.json:7]: `--build` takes multiple source directories as arguments. The previous construction of this command meant that the `--watch` argument from the `watch` script (`"npm run build -- --watch"`) only applied to `tsc --build scripts`.

**Review by sundersc — APPROVED** [review]

File: .changeset/perfect-coins-tease.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+'@aws-amplify/backend-data': minor
+'@aws-amplify/sandbox': minor
+'@aws-amplify/backend-cli': minor
+'@aws-amplify/backend-deployer': patch
+'@aws-amplify/backend-function': patch
+'@aws-amplify/client-config': patch
+'@aws-amplify/deployed-backend-client': patch
+'@aws-amplify/form-generator': patch
+'@aws-amplify/model-generator': patch
+'@aws-amplify/platform-core': patch
```
Comment (edwardfoyle) [inline .changeset/perfect-coins-tease.md:11]: We should remove all of these except `backend-data`. If we want to signal a change at the top level, that should just be the `@aws-amplify/backend` package, not these others. However, we haven't done that in the past so I'm not sure it's needed here either.

**Review by edwardfoyle — COMMENTED** [review]
functionally LGTM, just a comment on the changeset

File: .changeset/perfect-coins-tease.md (hunk 2 above)
Comment (palpatim) [reply .changeset/perfect-coins-tease.md:11]: I originally had only backend-data in the changeset (see [original commit](https://github.com/aws-amplify/amplify-backend/pull/1696/commits/d465f43295778474c206fd67a650151599cd67c5)), but the `health_checks / check_pr_changesets` check failed with a note about all those packages. I'm quite happy to revert the change, just let me know the preferred way of declaring it.

Comment (edwardfoyle) [reply .changeset/perfect-coins-tease.md:11]: Hmm, that check should only require changesets on touched packages. Maybe a merge from main would resolve the issue?

**Review by Amplifiyer — APPROVED** [review]

**Review by edwardfoyle — DISMISSED** [review]

**Comment by palpatim** [comment]
Converting to draft to wait for amplify-category-api release, to consume https://github.com/aws-amplify/amplify-category-api/pull/2680, which fixes the attribute name created by `ampx generate schema-from-database`

**Comment by palpatim** [comment]
amplify-category-api release is complete, ready to re-review

**Review by sundersc — APPROVED** [review]

**Review by Amplifiyer — APPROVED** [review]

**Pull request merged**